### PR TITLE
add experimental bootstrapper

### PR DESCRIPTION
**Pull request by roastario**

### CODE REVIEW

File: .idea/codeStyleSettings.xml [hunk 1]
```diff
@@ -1,19 +0,0 @@
-<?xml version="1.0" encoding="UTF-8"?>
```
Comment (tlil) [inline .idea/codeStyleSettings.xml]: Any particular reason for the deletion of all these config files?

File: tools/network-bootstrapper/src/main/kotlin/net/corda/bootstrapper/volumes/azure/AzureSmbVolume.kt [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+import net.corda.bootstrapper.volumes.Volume.Companion.networkMapCert
+import net.corda.core.internal.signWithCert
+import net.corda.core.serialization.serialize
+import net.corda.nodeapi.internal.network.NETWORK_PARAMS_FILE_NAME
+import org.slf4j.LoggerFactory
+
+
+class AzureSmbVolume(private val azure: Azure, private val context: Context) : Volume {
+
+    private val storageAccount = getStorageAccount()
```
[the remark this answers is not in the document]
Comment (roastario) [reply tools/network-bootstrapper/src/main/kotlin/net/corda/bootstrapper/volumes/azure/AzureSmbVolume.kt]: it actually automatically pauses the process and prints out instructions on how to reauthenticate. You then press "enter" and it continues if you have successfully authenticated your azure cli.

File: tools/network-bootstrapper/src/main/kotlin/net/corda/bootstrapper/cli/CommandParsers.kt [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+    }
+
+    override fun backendOptions(): Map<String, String> {
+        return mapOf(Constants.REGION_ARG_NAME to region.name())
+    }
+
+}
+
+
+fun main(badArgs: Array<String>) {
```
Comment (thschroeter) [inline tools/network-bootstrapper/src/main/kotlin/net/corda/bootstrapper/cli/CommandParsers.kt]: Remove this function?

**Comment by mikehearn** [comment]
> The usage help is not printed yet with -h and --help flags.

Can we make the tool use picocli? It does this automatically.

**Comment by tlil** [comment]
Stefano is already using picocli, so it's just hooking it up with the right call / set a flag. :)

**Comment by tlil** [comment]
@roastario as discussed, here comes a couple of suggestions / mockups re the UI:

image

image

image

What do you think? Makes it a bit easier to get an overview of what's happening, what's been created, built, published, etc. Essentially consolidates it into a single data grid instead of having different tables / lists of templates nodes/notaries to build, built nodes, published nodes, instances, etc.

File: tools/network-bootstrapper/src/main/kotlin/net/corda/bootstrapper/volumes/Volume.kt [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+    fun convertNodeIntoToNetworkParams(notaryFiles: List<Pair<File, File>>): NetworkParameters {
+        val notaryInfos = notaryFiles.map { (configFile, nodeInfoFile) ->
+            val validating = ConfigFactory.parseFile(configFile).getConfig("notary").getBoolean("validating")
+            nodeInfoFile.readBytes().deserialize<SignedNodeInfo>().verified().let { NotaryInfo(it.legalIdentities.first(), validating) }
+        }
+
+        return notaryInfos.let {
+            NetworkParameters(
+                    minimumPlatformVersion = 1,
```
[the remark this answers is not in the document]
Comment (tlil) [reply tools/network-bootstrapper/src/main/kotlin/net/corda/bootstrapper/volumes/Volume.kt]: Will you mark this as followup?

**Comment by roastario** [comment]
yes - that UI is much, much better. 

I'll get my artist set out and give it a go.

**Comment by tlil** [comment]
Cool, let me know if you want any input from me along the way :)

**Comment by tlil** [comment]
@roastario did you get around to doing anything about the UI design? or will that be addressed in a follow-up ticket?

**Pull request merged**